### PR TITLE
[PPDiffusers]upgrade_ppdiffusers0140

### PR DESCRIPTION
**Pull request by JunnYu**
### PR types

New features
### PR changes

PPDiffusers升级

### Description

升级ppdiffusers到0.14.0, 目前新增如下功能。

#### 加载Civitai社区的LoRA权重
```python
from ppdiffusers import StableDiffusionPipelineAllinOne
pipe = StableDiffusionPipelineAllinOne.from_pretrained("TASUKU2023/Chilloutmix")
# 加载lora权重
pipe.apply_lora("https://paddlenlp.bj.bcebos.com/models/community/junnyu/develop/ppdiffusers/Moxin_10.safetensors")
# 开启xformers
pipe.enable_xformers_memory_efficient_attention()
prompt = "# shukezouma, negative space, , shuimobysim , portrait of a woman standing , willow branches, (masterpiece, best quality:1.2), traditional chinese ink painting, <lora:shuimobysimV3:0.7>, modelshoot style, peaceful, (smile), looking at viewer, wearing long hanfu, hanfu, song, willow tree in background, wuchangshuo,"
negative_prompt = "(worst quality:2), (low quality:2), (normal quality:2), lowres, normal quality, skin spots, acnes, skin blemishes, age spot, glans, (watermark:2),"
img = pipe(prompt , negative_prompt=negative_prompt, num_inference_steps=50, height=512, width=512, guidance_scale=3.5).images[0]
img.save('lora.png')
```
image

关闭lora
```python
pipe.set_lora_enabled(False)
img = pipe(prompt , negative_prompt=negative_prompt, num_inference_steps=50, height=512, width=512, num_images_per_prompt=1, guidance_scale=3.5).images[0]
img.save("nolora.png")
```
image

保存lora
```python
pipe.save_lora("demo_lora")
```
image

#### 直接加载原版controlnet权重
```python
# 下载config yaml文件
!wget https://raw.githubusercontent.com/lllyasviel/ControlNet/main/models/cldm_v15.yaml
!wget https://raw.githubusercontent.com/lllyasviel/ControlNet/main/models/cldm_v21.yaml

from ppdiffusers import StableDiffusionControlNetPipeline, ControlNetModel, UniPCMultistepScheduler
from ppdiffusers.utils import load_image
import numpy as np
import paddle
import cv2
from PIL import Image

image = load_image(
     "https://hf.co/datasets/huggingface/documentation-images/resolve/main/diffusers/input_image_vermeer.png"
 )
image = np.array(image)
# get canny image
image = cv2.Canny(image, 100, 200)
image = image[:, :, None]
image = np.concatenate([image, image, image], axis=2)
canny_image = Image.fromarray(image)
# load control net and stable diffusion v1-5
ckpt = "https://paddlenlp.bj.bcebos.com/models/community/lllyasviel/ControlNet/control_sd15_canny.pth"
original_config_file = "cldm_v15.yaml"
kwargs = {"upcast_attention": False}
# ckpt = "https://paddlenlp.bj.bcebos.com/models/community/thibaud/controlnet-sd21/canny-sd21.ckpt"
# original_config_file = "cldm_v21.yaml"
# kwargs = {"upcast_attention": True}
pipe = StableDiffusionControlNetPipeline.from_pretrained_original_ckpt(
     ckpt,  paddle_dtype=paddle.float16,
    original_config_file=original_config_file,
    **kwargs,
 )
# 
# speed up diffusion process with faster scheduler and memory optimization
pipe.scheduler = UniPCMultistepScheduler.from_config(pipe.scheduler.config)
# remove following line if xformers is not installed
pipe.enable_xformers_memory_efficient_attention()
# generate image
generator = paddle.Generator().manual_seed(0)
image = pipe(
     "futuristic-looking woman", num_inference_steps=20, generator=generator, image=canny_image
 ).images[0]
image.save('canny.png')
```
image

```python
from ppdiffusers import StableDiffusionControlNetPipeline, ControlNetModel
import paddle

controlnet = ControlNetModel.from_pretrained("lllyasviel/sd-controlnet-canny", paddle_dtype=paddle.float16)
pipe = StableDiffusionControlNetPipeline.from_pretrained(
    "runwayml/stable-diffusion-v1-5", controlnet=controlnet, paddle_dtype=paddle.float16
)
```
#### 开启xformers
为了使用**XFormers加速**，我们需要安装`develop`版本的`paddle`，Linux系统的安装命令如下：
```sh
python -m pip install paddlepaddle-gpu==0.0.0.post117 -f https://www.paddlepaddle.org.cn/whl/linux/gpu/develop.html
```
```python
import paddle
from ppdiffusers import StableDiffusionPipeline
pipe = StableDiffusionPipeline.from_pretrained("TASUKU2023/Chilloutmix", paddle_dtype=paddle.float16)
# 开启xformers加速 默认选择"cutlass"加速
pipe.enable_xformers_memory_efficient_attention()
# flash 需要使用 A100、A10、3060、3070、3080、3090 等以上显卡。
# pipe.enable_xformers_memory_efficient_attention("flash")
```
速度比较 torch1.13.1 vs paddle develop 动态图 fp16 A100
image

#### 加载HF Diffusers权重
```python
from ppdiffusers import StableDiffusionPipeline
# 设置from_hf_hub为True，表示从huggingface hub下载，from_diffusers为True表示加载的是diffusers版Pytorch权重
pipe = StableDiffusionPipeline.from_pretrained("stabilityai/stable-diffusion-2", from_hf_hub=True, from_diffusers=True)
```
保存成HF的Diffusers权重, TIPS：当前text encoder可能无法保存成hf权重，需要额外注意。
```python
pipe.save_pretrained("hf_diffusers", to_diffusers=True)
```
#### DiTPipeline
```python
from ppdiffusers import DiTPipeline, DPMSolverMultistepScheduler
import paddle

pipe = DiTPipeline.from_pretrained("facebook/DiT-XL-2-512")
pipe.scheduler = DPMSolverMultistepScheduler.from_config(pipe.scheduler.config)
pipe.enable_xformers_memory_efficient_attention()
words = ["vase", "umbrella"]
ids = pipe.get_label_ids(words)
generator = paddle.Generator().manual_seed(0)
images = pipe(ids, generator=generator, num_inference_steps=25).images
images[0].save("dit.png")
```
image

#### StableDiffusionInstructPix2PixPipeline
```python
import paddle
from ppdiffusers import StableDiffusionInstructPix2PixPipeline
from ppdiffusers.utils import load_image

model_id = "timbrooks/instruct-pix2pix"
pipe = StableDiffusionInstructPix2PixPipeline.from_pretrained(model_id, from_diffusers=True, from_hf_hub=True)

url = "https://huggingface.co/datasets/diffusers/diffusers-images-docs/resolve/main/mountain.png"
image = load_image(url)

prompt = "make the mountains snowy"
edit = pipe(prompt, image=image, num_inference_steps=20, image_guidance_scale=1.5, guidance_scale=7).images[0]
edit.save("pix2pix.png")
```
image

#### DEIS Scheduler
```python
from ppdiffusers import StableDiffusionPipeline, DEISMultistepScheduler
import paddle

pipe = StableDiffusionPipeline.from_pretrained("runwayml/stable-diffusion-v1-5", paddle_dtype=paddle.float16)
pipe.scheduler = DEISMultistepScheduler.from_config(pipe.scheduler.config)

prompt = "a photo of an astronaut riding a horse on mars"
generator = paddle.Generator().manual_seed(40)
image = pipe(prompt, generator=generator, num_inference_steps=25).images[0]
image.save("deis.png")
```
image

#### Semantic guidance
```python
import paddle
from ppdiffusers import SemanticStableDiffusionPipeline

pipe = SemanticStableDiffusionPipeline.from_pretrained("runwayml/stable-diffusion-v1-5", paddle_dtype=paddle.float16)
out = pipe(
    prompt="a photo of the face of a woman",
    num_images_per_prompt=1,
    guidance_scale=7,
    editing_prompt=[
        "smiling, smile",  # Concepts to apply
        "glasses, wearing glasses",
        "curls, wavy hair, curly hair",
        "beard, full beard, mustache",
    ],
    reverse_editing_direction=[False, False, False, False],  # Direction of guidance i.e. increase all concepts
    edit_warmup_steps=[10, 10, 10, 10],  # Warmup period for each concept
    edit_guidance_scale=[4, 5, 5, 5.4],  # Guidance scale for each concept
    edit_threshold=[
        0.99,
        0.975,
        0.925,
        0.96,
    ],  # Threshold for each concept. Threshold equals the percentile of the latent space that will be discarded. I.e. threshold=0.99 uses 1% of the latent dimensions
    edit_momentum_scale=0.3,  # Momentum scale that will be added to the latent guidance
    edit_mom_beta=0.6,  # Momentum beta
    edit_weights=[1, 1, 1, 1, 1],  # Weights of the individual concepts against each other
 ).images[0]
out.save('semantic.png')
```
image

#### StableDiffusionLatentUpscalePipeline
```python
from ppdiffusers import StableDiffusionLatentUpscalePipeline, StableDiffusionPipeline
import paddle

pipeline = StableDiffusionPipeline.from_pretrained("CompVis/stable-diffusion-v1-4")

upscaler = StableDiffusionLatentUpscalePipeline.from_pretrained("stabilityai/sd-x2-latent-upscaler")

prompt = "a photo of an astronaut high resolution, unreal engine, ultra realistic"
generator = paddle.Generator().manual_seed(33)

# we stay in latent space! Let's make sure that Stable Diffusion returns the image
# in latent space
low_res_latents = pipeline(prompt, generator=generator, output_type="latent").images

upscaled_image = upscaler(
    prompt=prompt,
    image=low_res_latents,
    num_inference_steps=20,
    guidance_scale=0,
    generator=generator,
).images[0]

# Let's save the upscaled image under "upscaled_astronaut.png"
upscaled_image.save("astronaut_1024.png")

# as a comparison: Let's also save the low-res image
with paddle.no_grad():
    image = pipeline.decode_latents(low_res_latents)
image = pipeline.numpy_to_pil(image)[0]

image.save("astronaut_512.png")
```
image
image

#### Vae tiling
显存占用28G+
```python
import paddle
from ppdiffusers import StableDiffusionPipeline

pipe = StableDiffusionPipeline.from_pretrained("runwayml/stable-diffusion-v1-5", paddle_dtype=paddle.float16, safety_checker=None)
pipe.enable_xformers_memory_efficient_attention()
pipe.vae.enable_tiling()

prompt = "a beautiful landscape photo"
image = pipe(prompt, width=4096, height=2048, num_inference_steps=10).images[0]

image.save("4k_landscape.jpg")
```
image

#### StableDiffusionPix2PixZeroPipeline
```python
import requests
import paddle

from ppdiffusers import DDIMScheduler, StableDiffusionPix2PixZeroPipeline
from ppdiffusers.utils.testing_utils import load_pt

def download(embedding_url, local_filepath):
    r = requests.get(embedding_url)
    with open(local_filepath, "wb") as f:
        f.write(r.content)

def to_paddle(x):
    if hasattr(x, "numpy"):
        x = x.cpu().numpy()
    return paddle.to_tensor(x)

model_ckpt = "CompVis/stable-diffusion-v1-4"
pipeline = StableDiffusionPix2PixZeroPipeline.from_pretrained(model_ckpt, paddle_dtype=paddle.float16)
pipeline.scheduler = DDIMScheduler.from_config(pipeline.scheduler.config)

prompt = "a high resolution painting of a cat in the style of van gough"
source_emb_url = "https://hf.co/datasets/sayakpaul/sample-datasets/resolve/main/cat.pt"
target_emb_url = "https://hf.co/datasets/sayakpaul/sample-datasets/resolve/main/dog.pt"

for url in [source_emb_url, target_emb_url]:
    download(url, url.split("/")[-1])
import torch
src_embeds = to_paddle(torch.load(source_emb_url.split("/")[-1]))
target_embeds = to_paddle(torch.load(target_emb_url.split("/")[-1]))
images = pipeline(
     prompt,
     source_embeds=src_embeds,
     target_embeds=target_embeds,
     num_inference_steps=50,
     cross_attention_guidance_amount=0.15,
 ).images

images[0]
```
image

```python
import paddle
from paddlenlp.transformers import BlipForConditionalGeneration, BlipProcessor
from ppdiffusers import DDIMScheduler, DDIMInverseScheduler, StableDiffusionPix2PixZeroPipeline

import requests
from PIL import Image

captioner_id = "Salesforce/blip-image-captioning-base"
processor = BlipProcessor.from_pretrained(captioner_id)
model = BlipForConditionalGeneration.from_pretrained(
     captioner_id, 
 )
model.to(dtype=paddle.float16)
sd_model_ckpt = "CompVis/stable-diffusion-v1-4"
pipeline = StableDiffusionPix2PixZeroPipeline.from_pretrained(
     sd_model_ckpt,
     caption_generator=model,
     caption_processor=processor,
     paddle_dtype=paddle.float16,
     safety_checker=None,
 )

pipeline.scheduler = DDIMScheduler.from_config(pipeline.scheduler.config)
pipeline.inverse_scheduler = DDIMInverseScheduler.from_config(pipeline.scheduler.config)
pipeline.enable_model_cpu_offload()

img_url = "https://github.com/pix2pixzero/pix2pix-zero/raw/main/assets/test_images/cats/cat_6.png"

raw_image = Image.open(requests.get(img_url, stream=True).raw).convert("RGB").resize((512, 512))
# generate caption
caption = pipeline.generate_caption(raw_image)

# "a photography of a cat with flowers and dai dai daie - daie - daie kasaii"
inv_latents = pipeline.invert(caption, image=raw_image).latents
# we need to generate source and target embeds

source_prompts = ["a cat sitting on the street", "a cat playing in the field", "a face of a cat"]

target_prompts = ["a dog sitting on the street", "a dog playing in the field", "a face of a dog"]

source_embeds = pipeline.get_embeds(source_prompts)
target_embeds = pipeline.get_embeds(target_prompts)
# the latents can then be used to edit a real image
generator = paddle.Generator().manual_seed(42)
image = pipeline(
     caption,
     source_embeds=source_embeds,
     target_embeds=target_embeds,
     num_inference_steps=50,
     cross_attention_guidance_amount=0.15,
     generator=generator,
     latents=inv_latents,
     negative_prompt=caption,
 ).images[0]
image.save("edited_image.png")
```
image

### CODE REVIEW

File: ppdiffusers/ppdiffusers/models/modeling_utils.py [hunk 1]
```diff
@@ -192,85 +245,44 @@ def save_pretrained(
...
-                revision=revision,
-                create_pr=create_pr,
-            )
+        logger.info(f"Model weights saved in {os.path.join(save_directory, weights_name)}")
 
     @classmethod
     def from_pretrained(cls, pretrained_model_name_or_path: Optional[Union[str, os.PathLike]], **kwargs):
         r"""
-        Instantiate a pretrained paddle model from a pre-trained model configuration.
+        Instantiate a pretrained pytorch model from a pre-trained model configuration.
```
Comment (guoshengCS) [inline ppdiffusers/ppdiffusers/models/modeling_utils.py:285]: 这里是否有些漏掉的

File: ppdiffusers/ppdiffusers/webui_lora_helper.py [hunk 2]
```diff
@@ -0,0 +1,2622 @@
...
+    pipe_or_module,
+    lora_weight_or_path=None,
+    rank=4,
+    alpha=None,
+    multiplier=1.0,
+    text_encoder_target_replace_modules=["TransformerEncoderLayer"],
+    unet_target_replace_modules=["Transformer2DModel", "Attention"],
+    enable_lora=True,
+    **kwargs,
+):
```
Comment (guoshengCS) [inline ppdiffusers/ppdiffusers/webui_lora_helper.py:126]: 这些对用户暴露的比较重要的API后面估计还得加些API文档，比如说明这个`lora_weight_or_path`要是来源于pytorch的weubui的啥的

**Review by guoshengCS — APPROVED** [review]
LGTM

File: ppdiffusers/ppdiffusers/webui_lora_helper.py (hunk 2 above)
Comment (JunnYu) [reply ppdiffusers/ppdiffusers/webui_lora_helper.py:126]: OK
